SEED-platform/seed
# Wava Fixes

#### Any background context you want to provide?
Changes based on Netsparker security audit

#### What's this PR do?
1. AngularJS migrated to latest version

2. jQuery updated to latest version

3. Added `rel="noopener noreferrer"` to all external hyperlinks that specify `target="_blank"`

4. Added the `X-Content-Type-Options: nosniff` header, should show up on all responses served by Django but possibly not static files (consider adding this to Nginx)

5. Resolved mixed content over HTTPS by preventing Nginx from serving index.html files by default which could be accessed from the /static/ path

6. Added exclusions to `./manage.py collectstatic` to prevent copying package.json files and openlayers' index.html file.

7. Added `autocomplete="off"` to the login form

8. D3 updated to latest version

#### How should this be manually tested?
I recommend deleting your existing node_modules before continuing:
1. Delete `seed/node_modules`
2. Delete `seed/vendors/node_modules`

`npm install`

Monkey test, make sure all pages function normally (including pages with updated dependencies, like d3 for reports or openlayers for maps)

Also go through the list of 8 items and verify that each has been resolved
